go/oasis-test-runner/scenario/e2e/runtime/history_reindex.go

```diff
@@ -36,100 +37,112 @@ func (sc *historyReindexImpl) Fixture() (*oasis.NetworkFixture, error) {
+	f.Clients[0].Runtimes = nil
 
-func (sc *historyReindexImpl) Clone() scenario.Scenario {
-	return &historyReindexImpl{
-		Scenario: *sc.Scenario.Clone().(*Scenario),
-	}
+	return f, nil
 }
 
 func (sc *historyReindexImpl) Run(ctx context.Context, childEnv *env.Env) error {
-	cli := cli.New(childEnv, sc.Net, sc.Logger)
-
 	// Start the network.
 	if err := sc.Net.Start(); err != nil {
 		return err
 	}
 
-	// Wait for one block so that initial epoch can be fetched below.
-	if _, err := sc.WaitBlocks(ctx, 1); err != nil {
-		return fmt.Errorf("failed to wait for one block: %w", err)
+	// Prepare few runtime blocks for reindex.
+	compute := sc.Net.ComputeWorkers()[0]
+	computeCtrl, err := oasis.NewController(compute.SocketPath())
+	if err != nil {
+		return fmt.Errorf("failed to create controller: %w", err)
+	}
+	err = sc.waitRuntimeBlock(ctx, computeCtrl.RuntimeClient, 10)
+	if err != nil {
+		return fmt.Errorf("failed to wait for runtime block: %w", err)
+	}
+
+	// Reindex existing runtime blocks and start indexing new ones.
+	client := sc.Net.Clients()[0]
+	client.UpdateRuntimes([]int{sc.rtIdx})
+	if err = client.Restart(ctx); err != nil {
+		return fmt.Errorf("failed to restart %s: %w", client.Name, err)
 	}
 
-	// Restart compute worker with configured runtime.
-	compute := sc.Net.ComputeWorkers()[0]
-	sc.Logger.Info("stopping the compute worker")
-	if err := compute.Stop(); err != nil {
-		return err
+	// Verify that indexing works.
+	if err := sc.waitRuntimeBlock(ctx, sc.Net.ClientController().RuntimeClient, 20); err != nil {
+		return fmt.Errorf("failed to wait for runtime block: %w", err)
 	}
-	var rtIdx int
-	for idx, rt := range sc.Net.Runtimes() {
-		if rt.Kind() == registry.KindCompute {
-			rtIdx = idx
-			break
+
+	// Verify (re)indexed runtime blocks.
+	for round := uint64(0); round <= 20; round++ {
+		if err := sc.ensureEqualBlock(ctx, computeCtrl.RuntimeClient, sc.Net.ClientController().RuntimeClient, round); err != nil {
+			return fmt.Errorf("blocks not equal (round: %d): %w", round, err)
 		}
 	}
-	// Update worker runtime configuration.
-	compute.UpdateRuntimes([]int{rtIdx})
-	sc.Logger.Info("starting the compute worker")
-	if err := compute.Start(); err != nil {
-		return err
+
+	// Run test client to ensure runtime works.
+	sc.Logger.Info("starting the basic test client")
+	if err = sc.RunTestClientAndCheckLogs(ctx, childEnv); err != nil {
+		return fmt.Errorf("failed to run test client and check logs: %w", err)
 	}
 
-	// Fetch current epoch.
-	epoch, err := sc.Net.Controller().Beacon.GetEpoch(ctx, consensus.HeightLatest)
+	return nil
+}
+
+func (sc *historyReindexImpl) waitRuntimeBlock(ctx context.Context, client api.RuntimeClient, round uint64) error {
+	rtID := sc.Net.Runtimes()[sc.rtIdx].ID()
+	ch, sub, err := client.WatchBlocks(ctx, rtID)
 	if err != nil {
-		return fmt.Errorf("failed to get current epoch: %w", err)
+		return fmt.Errorf("client failed to watch runtime blocks (ID: %s)", rtID)
```

Comment:
```suggestion
		return fmt.Errorf("failed to watch runtime blocks")
```
Reader will have no idea which client are you talking about.